os_migrate/plugins/module_utils/network.py

```diff
@@ -94,24 +41,48 @@ def network_sdk_params(ser_net, net_refs):
         'provider_physical_network',
         'provider_segmentation_id',
         'segments',
-    ])
-    set_sdk_params_same_name(net_refs, sdk_params, [
+    ]
+    params_from_refs = [
+        'qos_policy_name',
+    ]
+    sdk_params_from_refs = [
         'qos_policy_id',
-    ])
-
-    return sdk_params
-
-
-def network_needs_update(sdk_net, net_refs, target_ser_net):
-    """Having OpenStack SDK network `sdk_net` and corresponding id-name
-    mappings `net_refs`, decide if the network needs to be updated to
-    reach state represented in OS-Migrate Network serialization
-    `target_ser_net`.
-
-    Returns: True if network needs to be updated, False otherwise
-    """
-    current_ser_net = serialize_network(sdk_net, net_refs)
-    return serialization.resource_needs_update(current_ser_net, target_ser_net)
+    ]
+
+    @classmethod
+    def from_sdk(cls, conn, sdk_resource):
+        obj = super(Network, cls).from_sdk(conn, sdk_resource)
+        obj._sort_param('availability_zone_hints')
+        obj._sort_info('subnet_ids')
+        return obj
+
+    def create_or_update(self, conn):
+        refs = self._refs_from_ser(conn)
+        sdk_params = self._to_sdk_params(refs)
+        existing = conn.network.find_network(sdk_params['name'])
```

Comment:
DRYing this into Resource class can come in a separate commit. We cannot simply make a variable for something like `find_network` here, pointing to a static method reference in OpenStack. We'll need to look up that method dynamically on the run-time `conn` object. So DRYing this logic will be more involved and require using intra-language reflection (looking up methods by name).

We can decide separately if we want to introduce this amount of magic or not for making things DRY, but i think this PR already improves things a lot and is worth merging even without this part done.